python/ambassador/envoy/v3/v3cluster.py

```diff
@@ -74,6 +74,18 @@ def __init__(self, config: "V3Config", cluster: IRCluster) -> None:
         if cluster.respect_dns_ttl:
             fields["respect_dns_ttl"] = cluster.respect_dns_ttl
 
+        # Don't feed the raw health_checks object to envoy, grab its config where the econf has already been built
+        if cluster.health_checks is not None:
+            health_checks_config = cluster.health_checks.config()
+            if health_checks_config is not None:
+                if ctype == "EDS":
+                    self["health_checks"] = health_checks_config
+                else:
+                    cluster.ir.logger.error(
+                        "health_checks configuration is only supported when using the endpoint resolver. Ignoring health_checks config for cluster: %s..."
+                        % (cluster.name)
```

Comment:
Is the cluster name going to be sufficient for the user to figure out which one is causing the error? Are we able to provide additional information here? I don't have any thing off the top of my head but just wanted to ask?

Reply:
Not sure what else we would be able to provide at this point for debugging unfortunately.